src/components/VirtualThought.tsx

```diff
@@ -153,7 +158,7 @@ const VirtualThought = ({
       isVisible: isVisibleNew,
       key: crossContextualKey,
     })
-  }, [crossContextualKey, onResize, path, thought.id])
+  }, [crossContextualKey, onResize, path, thought.id, noOtherContexts])
```

Comment:
It's not quite right to add `noOtherContexts` as a dependency to the `updateSize` definition, since the `updateSize` function does not depend on it. It would be better to add it to the `useEffect` hook that *calls* `updateSize`.

The problem is indirectly solved with the current approach because `updateSize` itself is a dependency of the `useEffect` hook that calls `updateSize`.

Reply:
Yes, that makes sense. 
I've pushed the update.